subprojects/docs/src/docs/userguide/authoring-builds/build_lifecycle.adoc

```diff
@@ -57,76 +73,81 @@ include::{snippetsPath}/buildlifecycle/basic/tests/buildlifecycle.out[]
+Every project except the root project has a parent project.
+Any project may have child projects.
 
-Your build script can receive notifications as the build progresses through its lifecycle. These notifications generally take two forms: You can either implement a particular listener interface, or you can provide a closure to execute when the notification is fired. The examples below use closures. For details on how to use the listener interfaces, refer to the API documentation.
+== Configuration
 
+In the configuration phase, Gradle adds tasks and other properties to the projects generated by the initialization phase.
+By the end of the configuration phase, Gradle has a complete task execution graph for each project.
 
 [[sec:project_evaluation]]
-=== Project evaluation
+=== Project Evaluation
+
+During project evaluation, Gradle evaluates build scripts to build a task hierarchy within a `Project`.
+This hierarchy includes inputs, actions, and outputs for all tasks.
 
-You can receive a notification immediately before and after a project is evaluated. This can be used to do things like performing additional configuration once all the definitions in a build script have been applied, or for some custom logging or profiling.
+==== React to Project Evaluation
 
-Below is an example which adds a `test` task to each project which has a `hasTests` property value of true.
+You can receive a notification immediately before and immediately after a project evaluates.
+These notifications work even when project evaluation fails.
+You can configure project evaluation notifications for all projects or a specific project.
+For example, you could use these notifications for:
+
+- additional configuration after applying all definitions in a build script
+- custom logging
+- custom profiling
+
+The following example adds a `test` task to each project where the `hasTests` property value is `true`:
 
-.Adding of test task to each project which has certain property set
 ====
 include::sample[dir="snippets/buildlifecycle/projectEvaluateEvents/groovy",files="build.gradle[tags=after-evaluate];project-a.gradle[]"]
 include::sample[dir="snippets/buildlifecycle/projectEvaluateEvents/kotlin",files="build.gradle.kts[tags=after-evaluate];project-a.gradle.kts[]"]
 ====
 
-.Output of `gradle -q test`
 ----
 > gradle -q test
 include::{snippetsPath}/buildlifecycle/projectEvaluateEvents/tests/projectEvaluateEvents.out[]
 ----
 
-This example uses method `Project.afterEvaluate()` to add a closure which is executed after the project is evaluated (ie. the end of the configuration phase).
-
-It is also possible to receive notifications when any project is evaluated. This example performs some custom logging of project evaluation. Notice that the `afterProject` notification is received regardless of whether the project evaluates successfully or fails with an exception.
+The following example uses `Project.afterEvaluate()` to run logic when a project finishes evaluation:
```

Comment:
The example actually doesn't use `afterEvaluate {}` which BTW is discouraged. It uses `gradle.afterProject {}` instead.

Reply:
```suggestion
The following example uses `Project.afterProject()` to run logic when a project finishes evaluation:
```
Should we then rephrase to this?

Reply:
Yes :+1: